modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -259,18 +259,44 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
         if groupby_args["level"] is not None:
             raise NotImplementedError("levels are not supported for groupby")
 
-        groupby_cols = by_frame.columns.tolist()
-        agg_cols = [col for col in self.columns if col not in by_frame.columns]
+        groupby_cols = by_frame.columns
+        if isinstance(agg, dict):
+            agg_cols = agg.keys()
```

Comment:
this feels like it fixes some separate issue... if that is so please file a separate PR

Reply:
this is also about incorrect handling of duplicated columns, if I revert this change newly added tests will fail